### PR TITLE
Fixes #8428 - Connecting audits to existing users

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

File: test/unit/concerns/audit_extensions_test.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class AuditExtensionsTest < ActiveSupport::TestCase
+  def setup
+    User.current = users :admin
```
Comment (domcleal) [inline test/unit/concerns/audit_extensions_test.rb]: please avoid setting User.current, use the `as_admin` helper

**Comment by orrabin** [comment]
tests are failing with the error: 

`Cannot delete or update a parent row: a foreign key constraint fails (`test_develop_pr_core-3-test`.`audits`, CONSTRAINT`audits_user_id_fk`FOREIGN KEY (`user_id`) REFERENCES`users`(`id`)): DELETE FROM`users`WHERE`users`.`id`= 135138680`

it looks related to the fact that audit-user association is polymorphic:
https://rails.lighthouseapp.com/projects/8994/tickets/974-attribute_type-is-not-set-to-null-if-dependent-nullify-is-specified-for-a-belongs_to-association

even after trying to override the association declaration and adding `:polymorphic => false` the error continues

File: test/unit/concerns/audit_extensions_test.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require 'test_helper'
+
+class AuditExtensionsTest < ActiveSupport::TestCase
+
+  def setup
+    @user =  users :admin
```
Comment (dLobatog) [inline test/unit/concerns/audit_extensions_test.rb:6]: There's an extra whitespace here, I'll take care of it on merge

**Comment by dLobatog** [comment]
Works well, merging in 3..2..1... :bomb:

**Comment by dLobatog** [comment]
Merged as 642c9d99a4c9127765cdca65f138eed12b7371f8 , thanks @orrabin !

**Pull request closed without merging**